ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/tree/TomlValue.kt

```diff
@@ -175,6 +257,31 @@ internal constructor(
             }
             return nbUnicodeChars
         }
+
+        private fun String.escapeSpecialCharacters(): String {
+            val withCtrlCharsEscaped = replace(controlCharacterRegex) { match ->
+                when (val char = match.value.single()) {
+                    '\b' -> "\\b"
+                    '\n' -> "\\n"
+                    '\u000C' -> "\\f"
+                    '\r' -> "\\r"
+                    else -> {
+                        val code = char.code
+
+                        val hexDigits = code.toString(HEX_RADIX)
+
+                        "\\$SIMPLE_UNICODE_PREFIX${
+                            hexDigits.padStart(SIMPLE_UNICODE_LENGTH, '0')
```

Comment:
I think if you only use the `\u` prefix, then it's going to be problematic for complex unicode symbols. For example, 🔥 is `U+1F525`, and would be written as `\U0001F525` in TOML. A simple method would just be to check `hexDigits` length and see which prefix to use

Reply:
This code is really only supposed to touch control characters though, which are well within the `\u` range

Reply:
oh alright, my bad, didn't understand that part properly